qdb/utils.py

```diff
@@ -123,49 +134,52 @@ def __str__(self):
 
     def __repr__(self):
         return 'QdbTimeout(seconds=%s, exception=%s, timer_signal=%s)' \
-            % (self.seconds, self.exception, signal_module.SIGALRM)
+            % (self.seconds, self._exception, signal_module.SIGALRM)
 
 
-class _TimeoutMagic(tuple):
-    """
-    _TimeoutMagic is really just a tuple that can be called to get a new
-    Timeout that is either gevented or not.
-    """
-    def __call__(self, seconds, exception=None, green=False):
+if gevent is not None:
+    class _TimeoutMagic(tuple):
```

Comment:
FWIW, this is precisely the reason that virtual subclasses are part of ABCMeta.
https://docs.python.org/2/library/abc.html#abc.ABCMeta.register